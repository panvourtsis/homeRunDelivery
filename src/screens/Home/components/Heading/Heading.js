import React from 'react'
import PropTypes from 'prop-types'
import { View, Text, TouchableOpacity } from 'react-native'
import Feather from 'react-native-vector-icons/Feather'
import styles from './Heading.style'

const Heading = ({title, onPress}) => (
  <View style={styles.container}>
    <Text style={styles.heading}>{title}</Text>

    <TouchableOpacity
      style={styles.viewMoreContainer}
      onPress={onPress}>
      <Text style={styles.viewMore}>View More</Text>
      <Feather
        style={styles.viewMore}
        name='chevron-right'
        size={22}
      />
    </TouchableOpacity>
  </View>
)

Heading.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired
}

export default Heading
