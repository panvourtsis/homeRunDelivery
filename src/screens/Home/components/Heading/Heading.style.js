const styles = {
  container: {flexDirection: 'row', justifyContent: 'space-between', paddingTop: 20, paddingBottom: 5, paddingLeft: 15, paddingRight: 15},
  heading: {
    color: '#7ca549',
    fontWeight: 'bold'
  },
  viewMoreContainer: {flexDirection: 'row', alignSelf: 'flex-end'},
  viewMore: {
    color: '#767676',
    paddingLeft: 5
  }
}

export default styles
