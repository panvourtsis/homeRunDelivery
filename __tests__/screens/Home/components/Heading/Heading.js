import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import Heading from '@screens/Home/components/Heading/Heading'

describe('Testing Heading', () => {
  let wrapper
  const spyFn = jest.fn()
  beforeEach(() => {
    wrapper = shallow(
      <Heading title={'Fresh Food'} onPress={spyFn} />
    )
  })
  it('renders as expected', () => {
    expect(wrapper.dive()).toMatchSnapshot()
  })
  it('onPress it triggers function', () => {
    wrapper.dive().find('TouchableOpacity').simulate('press')
    expect(spyFn).toHaveBeenCalledTimes(1)
  })
})
