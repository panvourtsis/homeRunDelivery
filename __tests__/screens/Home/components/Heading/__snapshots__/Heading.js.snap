// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Heading renders as expected 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 5,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#7ca549",
        "fontWeight": "bold",
      }
    }
  >
    Fresh Food
  </Text>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[MockFunction]}
    style={
      Object {
        "alignSelf": "flex-end",
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#767676",
          "paddingLeft": 5,
        }
      }
    >
      View More
    </Text>
    <Icon
      allowFontScaling={false}
      name="chevron-right"
      size={22}
      style={
        Object {
          "color": "#767676",
          "paddingLeft": 5,
        }
      }
    />
  </TouchableOpacity>
</View>
`;
